src/serlo-editor/plugins/text/components/headless-text-editor.tsx

```diff
@@ -0,0 +1,176 @@
+import isHotkey from 'is-hotkey'
+import React, { useRef, useMemo, useState, useCallback } from 'react'
+import { Descendant, Node, Transforms, Range, Editor } from 'slate'
+import { Editable, RenderElementProps, Slate } from 'slate-react'
+
+import { LinkControls } from './link/link-controls'
+import { MathElement } from './math-element'
+import { TextLeafRenderer } from './text-leaf-renderer'
+import { useFormattingOptions } from '../hooks/use-formatting-options'
+import { useTextConfig } from '../hooks/use-text-config'
+import { TextEditorPluginConfig, TextEditorState } from '../types'
+import { useEditorStrings } from '@/contexts/logged-in-data-context'
+import { EditorPluginProps } from '@/serlo-editor/plugin'
+
+export interface HeadlessTextEditorConfig {
+  placeholder?: TextEditorPluginConfig['placeholder']
+  noLinebreaks?: boolean
+  serloLinkSearch: boolean
+  controls: {
+    editor: Editor
+    textFormattingOptions: ReturnType<typeof useFormattingOptions>
+  }
+}
+
+export type HeadlessTextEditorProps = EditorPluginProps<
+  TextEditorState,
+  HeadlessTextEditorConfig
+>
+
+// TODO: Think of a better name, "headless" doesn't really make sense.
+// Something like "ChildTextEditor" or similar
```

Comment:
Have not thought it through, but so far we only use it for inline texts. Maybe `InlineTextEditor` would work?